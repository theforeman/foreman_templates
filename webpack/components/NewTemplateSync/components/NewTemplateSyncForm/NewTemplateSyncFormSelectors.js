import { createSelector } from 'reselect';

import {
  selectImportSettings,
  selectExportSettings,
} from '../../NewTemplateSyncSelectors';

export const transformInitialValues = settings =>
  settings.reduce(
    (memo, item) => Object.assign(memo, { [item.name]: item.value }),
    {}
  );

export const selectInitialFormValues = createSelector(
  selectImportSettings,
  selectExportSettings,
  (importSettings, exportSettings) => ({
    import: transformInitialValues(importSettings),
    export: transformInitialValues(exportSettings),
  })
);
