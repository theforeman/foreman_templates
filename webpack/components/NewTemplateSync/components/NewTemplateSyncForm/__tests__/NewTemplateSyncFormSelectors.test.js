import { testSelectorsSnapshotWithFixtures } from 'react-redux-test-utils';

import {
  stateFactory,
  importSettings,
  exportSettings,
} from '../../../__fixtures__/templateSyncSettings.fixtures';

import { selectInitialFormValues } from '../NewTemplateSyncFormSelectors';

const fixtures = {
  'should return initial form values': () =>
    selectInitialFormValues({
      ...stateFactory({ importSettings, exportSettings }),
    }),
};

describe('NewTemplateSyncFormSelectors', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
