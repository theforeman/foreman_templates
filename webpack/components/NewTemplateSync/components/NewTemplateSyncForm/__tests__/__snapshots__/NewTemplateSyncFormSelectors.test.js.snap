// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTemplateSyncFormSelectors should return initial form values 1`] = `
Object {
  "export": Object {
    "filter": "",
    "negate": false,
    "repo": "https://github.com/theforeman/community-templates.git",
  },
  "import": Object {
    "associate": "new",
    "force": false,
  },
}
`;
